daemon/config.go

```diff
@@ -121,6 +125,10 @@ type CommonConfig struct {
 	// may take place at a time for each push.
 	MaxConcurrentUploads *int `json:"max-concurrent-uploads,omitempty"`
 
+	// ShutdownTimeout is the timeout value (in seconds) the daemon will wait for the container
+	// to stop when daemon is being shutdown
+	ShutdownTimeout int `json:"shutdown-timeout,omitempty"`
```

Comment:
@cpuguy83 One potential issue with `time.Duration` is that the default marshal/unmarshal of `time.Duration` is nano-seconds (cannot parse a string like "5s"). In other words, if user wants to specify 5 seconds, it will be `"shutdown-timeout": 5000000000`, which might not be ideal.

Maybe we could use a customized type, e.g., 
```
type Duration struct {
    time.Duration
}
func (d Duration) MarshalJSON() ([]byte, error) {
```
so that user could specify `5s` instead of `5000000000`?